File: tmp_claude_ws_probe.py
import asyncio, json
import websockets

async def main():
    async with websockets.connect('ws://127.0.0.1:8081/ws?token=test') as ws:
        for _ in range(2):
            print('INIT', await ws.recv())
        await ws.send(json.dumps({'action': 'exec', 'mode': 'pty', 'cmd': 'claude'}))
        sent = False
        for i in range(30):
            msg = await asyncio.wait_for(ws.recv(), timeout=20)
            print('EVT', msg)
            data = json.loads(msg)
            if not sent and data.get('type') == 'agent_state' and data.get('state') == 'WAIT_INPUT':
                await ws.send(json.dumps({'action': 'input', 'data': '你好\n'}))
                print('SENT_INPUT')
                sent = True

asyncio.run(main())

File: tmp_claude_ws_probe.js
const wsUrl = "ws://127.0.0.1:8081/ws?token=test";
const cmd = "claude";

if (typeof WebSocket === "undefined") {
  console.error("当前 Node.js 没有内置 WebSocket。");
  process.exit(1);
}

const ws = new WebSocket(wsUrl);
let inputSent = false;
let count = 0;

ws.addEventListener("open", () => {
  console.log("OPEN", wsUrl);
  ws.send(JSON.stringify({ action: "exec", mode: "pty", cmd }));
});

ws.addEventListener("message", (evt) => {
  const raw = typeof evt.data === "string" ? evt.data : String(evt.data);
  console.log("EVT", raw);
  let event;
  try { event = JSON.parse(raw); } catch { return; }
  count += 1;
  if (!inputSent && event.type === "agent_state" && event.state === "WAIT_INPUT") {
    inputSent = true;
    ws.send(JSON.stringify({ action: "input", data: "你好\n" }));
    console.log("SENT_INPUT");
  }
  if (count > 30) {
    ws.close();
    process.exit(0);
  }
});

ws.addEventListener("close", () => {
  console.log("CLOSED");
  process.exit(0);
});

ws.addEventListener("error", (err) => {
  console.error("WS_ERROR", err);
  process.exit(2);
});

setTimeout(() => {
  console.error("TIMEOUT");
  try { ws.close(); } catch {}
  process.exit(3);
}, 30000);
